main.go

```diff
@@ -116,6 +126,61 @@ func main() {
 		os.Exit(1)
 	}
 
+	if createGrpc {
+		// TODO: should this be one go routine or several?
+		g, ctx := errgroup.WithContext(ctx)
+		g.Go(func() error {
+			k8sClient := mgr.GetClient()
+			// GET deployment to see if it exists first before creating it
+			deployment := &appsv1.Deployment{}
+			err := k8sClient.Get(ctx, types.NamespacedName{Name: controllers.GrpcDeployment.Name, Namespace: controllers.GrpcDeployment.Namespace}, deployment)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					setupLog.Info("creating grpc deployment")
+					return k8sClient.Create(ctx, controllers.GrpcDeployment, &client.CreateOptions{})
+				}
+			}
+			// NOTE: Don't crash, just don't deploy if Get fails for any other reason than not found
+			return nil
+		})
+
+		g.Go(func() error {
+			k8sClient := mgr.GetClient()
+			service := &corev1.Service{}
+			err := k8sClient.Get(ctx, types.NamespacedName{Name: controllers.GrpcService.Name, Namespace: controllers.GrpcService.Namespace}, service)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					setupLog.Info("creating grpc service")
+					return k8sClient.Create(ctx, controllers.GrpcService, &client.CreateOptions{})
+				}
+			}
+			// NOTE: Don't crash, just don't deploy if Get fails for any other reason than not found
+			return nil
+		})
+
+		g.Go(func() error {
+			k8sClient := mgr.GetClient()
+			cm := &corev1.ConfigMap{}
+			err := k8sClient.Get(ctx, types.NamespacedName{Name: controllers.GrpcConfigMap.Name, Namespace: controllers.GrpcConfigMap.Namespace}, cm)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					setupLog.Info("creating grpc configmap")
+					return k8sClient.Create(ctx, controllers.GrpcConfigMap, &client.CreateOptions{})
+				}
+			}
+			// NOTE: Don't crash, just don't deploy if Get fails for any other reason than not found
+			return nil
+		})
+
+		go func() {
+			if err := g.Wait(); err != nil {
+				setupLog.Error(err, "error with async operation of creating grpc deployment")
+				os.Exit(1)
+			}
+			setupLog.Info("grpc server has been created")
+		}()
+	}
```

Comment:
This is dependent on the resources like configmap being there first, etc.  So the ordering of these async methods may need to be:
1. Check/Create configmap
2. Check/Create service
3. Check/Create deployment

So this will probably being one method.  

Also, this required more from an RBAC perspective.

Reply:
When the operator starts, the cache is not loaded to GET any resource.  The create will have to happen and if it exists (conflict), we will assume this will be there.  

This doesn't ensure that someone will not delete the deployment/configmap/service.

Reply:
Resolving.